### ISSUE TITLE
Support for managed identity in azure.storage.fileshare

### ISSUE THREAD

**Issue by holwech** [Azure/azure-sdk-for-python#13033]
Running the following:

```python
from azure.storage.fileshare import ShareFileClient
from azure.identity import DefaultAzureCredential

file_client = ShareFileClient(
                        account_url=f"https://{storage_account}.file.core.windows.net/",
                        file_path=file_path,
                        credential=DefaultAzureCredential(),
                        share_name=share_name
                     )
```
Leads to the following error:

```python
ValueError("Token credentials not supported by the File service.")
```

Default credentials in my case would be managed identity. It's not explicitly mentioned in the documentation (as far as I can see), but is managed identity not supported by this library? Will it be supported, and if so, when?

**Comment by xiafu-msft** [comment]
Hi @holwech
Thanks for reporting this. We will discuss this and get back to you!

**Comment by bec-zt8** [comment]
what is the status of this issue?

**Comment by amishra-dev** [comment]
The service does not have the support for this yet. We will add it to the SDK as soon as the service adds support. We do not have the ETA for the service support

**Comment by pdworzynski** [comment]
Hi, 

Is there an update on this? For example, [documentation](https://docs.microsoft.com/en-us/python/api/overview/azure/storage-file-share-readme?view=azure-python) suggests that this approach should work e.g.

> ```py
> from azure.storage.fileshare import ShareServiceClient
> 
> service = ShareServiceClient(account_url="https://<my-storage-account-name>.file.core.windows.net/", credential=credential)
> ```

However, using azure-storage-file-share 12.5.0 with a service principal token i'm getting the same error.
```py 
ValueError("Token credentials not supported by the File service.") 
``` 

Thanks!

**Comment by jvschoen** [comment]
Any update on this? what is the best approach to connecting to file service through python SDK?

**Comment by jalauzon-msft** [comment]
Hi all, apologies for the long delay with no update but we've essentially been waiting for the service to add support for AAD auth to the Files service. I'm happy to report the service team is finally working on adding this support and we should soon be able to support this in the SDK. I don't have yet have a concrete timeline but hopefully it will be coming within the next couple of releases.

Thanks for your patience. I will update here once the feature has been added.

**Comment by david-msft** [comment]
> Hi all, apologies for the long delay with no update but we've essentially been waiting for the service to add support for AAD auth to the Files service. I'm happy to report the service team is finally working on adding this support and we should soon be able to support this in the SDK. I don't have yet have a concrete timeline but hopefully it will be coming within the next couple of releases.
> 
> Thanks for your patience. I will update here once the feature has been added.

is this going to be in .net SDK as well?

**Comment by jalauzon-msft** [comment]
@david-msft, yes, the plan is for the to be added to the .NET SDK as well once its available from the service. Still no ETA on it but this feature in.NET should be released at the same as Python.

**Comment by paul-cleverley** [comment]
what @diti-nz said...?

**Comment by jalauzon-msft** [comment]
Hi all, I'm happy to announce that support for Files OAuth / Managed identities has been released in our latest preview release, [12.12.0b1](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/storage/azure-storage-file-share/CHANGELOG.md#12120b1-2023-03-28)! Please see [this sample](https://github.com/Azure/azure-sdk-for-python/blob/28e59661077a616f97b8badc6b6a5cec59da4e55/sdk/storage/azure-storage-file-share/samples/file_samples_authentication.py#L75-L98) for example use.

Please note there some caveats with the implementation that are specific to the Files SDK and come from the service (out of the client's control).
- Only File, Directory, and a small handful of Share level operations are supported for use with OAuth in the data plane SDK (`azure-storage-file-share`). For all other operations that need to be used with OAuth, it is recommended to use the control plane SDKs.
- In the current implementation you **must** provide the `token_intent` keyword (see sample) with the only possible value being `backup` today. This certifies your intent to use the `readFileBackupSemantics` and/or `writeFileBackupSemantics` RBAC permissions that allow you to bypass File ACLs when using OAuth. This is currently the only supported scenario for OAuth in the Files service.

Please feel free to try out the preview release and provide any feedback. This will tentatively be released in our next full release within the next few weeks (may change though). Since it is currently in preview, it is subject to change before the full release though. Thanks for your patience!

**Comment by jalauzon-msft** [comment]
This has been fully released in version [12.12.0](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/storage/azure-storage-file-share/CHANGELOG.md#12120-2023-04-12).
